docs/framework/concepts/yaml_format.md

```diff
@@ -0,0 +1,283 @@
+      slots:
+        content:
+          type: Pager.Simple
+          slots:
+            home-page:
+              shortdef: 'Layout.InfiniteScrolling(expand: true)'
+              references:
+                lazy-load: all-articles
+              slots:
+                content:
+                  shortdef: 'ContentGroup.ContentGroup(expand: true)'
+                  slots:
+                    arrangement:
+                      shortdef: 'Arrangement.List(expand: true)'
+                      slots:
+                        card: 'Card.List(expand: true)'
+                    selection:
+                      type: Selection.All
+                      id: all-articles
+                      slots:
+                        filter: Filter.Articles
+                        order: Order.Sequence
+```
+
+In this example a _Layout.InfiniteScrolling_ module is added, which needs access to the _Selection.All_ module. To do this, the selection module defines its identifier "all-articles" using the `id` keyword. Then, the scrolling module `references` the selection using the identifier. The "lazy-load" reference name is what the scrolling module uses to access this reference. The list of available references names depends on each type of module.
+
+## Styles
+
+App creators will likely need to style specific nodes, and to avoid needing complex CSS selectors, the easiest way to do this is by adding custom CSS classes. The format includes the `styles` keyword to add one or more CSS classes. See the example below:
+
+```
+---
+root:
+  type: Controller.Mesh
+  slots:
+    window:
+      type: Window.Simple
+      slots:
+        content:
+          type: Pager.Simple
+          slots:
+            home-page:
+              shortdef: 'Layout.InfiniteScrolling(expand: true)'
+              references:
+                lazy-load: all-articles
+              slots:
+                content:
+                  shortdef: 'ContentGroup.ContentGroup(expand: true)'
+                  styles:
+                    - ContentGroup--articles
+                  slots:
+                    arrangement:
+                      shortdef: 'Arrangement.List(expand: true)'
+                      slots:
+                        card: 'Card.List(expand: true)'
+                    selection:
+                      type: Selection.All
+                      id: all-articles
+                      slots:
+                        filter: Filter.Articles
+                        order: Order.Sequence
+```
+
+In this example, a "ContentGroup--articles" CSS class is added to a node, after that the CSS class can be used from the app's theme to style that specific node.
+
+## Imports
+
+Each app must include its own app description, but there is no need to write it from scratch. The framework provides pre-defined apps descriptions or presets, for different kind of app experiences. The format includes the `!import` keyword to use these presets. See the example below:
```

Comment:
There is a name in the YAML terminology for `!something`, but I forget what it is.
different kind -> different kinds